crates/core/tedge_api/src/messages.rs

```diff
@@ -616,6 +616,47 @@ impl ConfigUpdateCmdPayload {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct FirmwareMetadata {
+    pub name: Option<String>,
+    pub version: Option<String>,
```

Comment:
Unless if the user does not know the name/version, or they would rather just use whatever name/version is actually contained inside the firmware image linked to the give url. Otherwise you would have to be sure that the name/version actually matched the whatever binary is behind the url.

I don't mind if name/version is optional (even though Cumulocity will not allow you to do this from the UI). But the local tedge interface should not be so strict as for us it does not really matter.